DependencyInjection/Compiler/DbalSchemaFilterPass.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Processes the doctrine.dbal.schema_filter
+ */
+class DbalSchemaFilterPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        // feature require doctrine/dbal 2.9 or higher
+        if (!method_exists(\Doctrine\DBAL\Configuration::class, 'setSchemaAssetsFilter')) {
+            return;
+        }
+
+        $filters = $container->findTaggedServiceIds('doctrine.dbal.schema_filter');
+
+        $connectionFilters = [];
+        foreach ($filters as $id => $tagAttributes) {
+            foreach ($tagAttributes as $attributes) {
+                $name = isset($attributes['connection']) ? $attributes['connection'] : $container->getParameter('doctrine.default_connection');
```

Comment:
should ignoring the name apply it to the default connection, or to all connections ?

Reply:
I was wondering about this too. Making it apply to all connections is actually more useful - as, for example, Symfony's session system could (in theory) say: "Hey, ignore a `session` table that exits in any connection... I'm not sure which connection is used for the PDO instance I'm using".

But, that also seems like a possible, legitimate edge-case: I want to ignore `session` on connection 1, but I really DO have a `session` table on connection2 that I'm managing.

Reply:
I agree with Ryan here: users should know which connection handles which tables and can build their ignore lists accordingly.